tests/components/ps4/test_media_player.py

```diff
@@ -0,0 +1,469 @@
+
+    assert hass.states.get(mock_entity_id).state == STATE_IDLE
+
+
+async def test_state_none_is_set(hass):
+    """Test that state is set to None."""
+    mock_entity_id = await setup_mock_component(hass)
+
+    assert hass.states.get(mock_entity_id).state == STATE_UNKNOWN
+
+
+async def test_media_attributes_are_fetched(hass):
+    """Test that media attributes are fetched."""
+    mock_entity_id = await setup_mock_component(hass)
+    mock_device = hass.data[PS4_DATA].devices[0]
+    mock_func = '{}{}'.format('homeassistant.components.ps4.media_player.',
+                              'pyps4.Ps4Async.async_get_ps_store_data')
+    assert not mock_device.source_list
+
+    # Mock result from fetching data.
+    mock_result = MagicMock()
+    mock_result.name = MOCK_TITLE_NAME
+    mock_result.cover_art = MOCK_TITLE_ART_URL
+    mock_result.game_type = 'game'
+
+    with patch(mock_func, return_value=mock_coro(mock_result)) as mock_fetch,\
+            patch(MOCK_SAVE, side_effect=MagicMock()):
+
+        await mock_ddp_response(hass, MOCK_STATUS_PLAYING)
+
+    mock_state = hass.states.get(mock_entity_id).state
+
+    assert len(mock_fetch.mock_calls) == 1
+
+    assert mock_state == STATE_PLAYING
+    assert len(mock_device.source_list) == 1
+    assert mock_device.media_content_id == MOCK_TITLE_ID
+    assert mock_device.media_title == MOCK_TITLE_NAME
+    assert mock_device.media_image_url == MOCK_TITLE_ART_URL
+    assert mock_device.media_content_type == MOCK_TITLE_TYPE
+
+
+async def test_media_attributes_are_loaded(hass):
+    """Test that media attributes are loaded."""
+    mock_entity_id = await setup_mock_component(hass)
+    mock_device = hass.data[PS4_DATA].devices[0]
+    mock_data = {MOCK_TITLE_ID: MOCK_GAMES_DATA_LOCKED}
+    mock_func = '{}{}'.format('homeassistant.components.ps4.media_player.',
+                              'pyps4.Ps4Async.async_get_ps_store_data')
+
+    with patch(mock_func, return_value=mock_coro(None)) as mock_fetch:
+        await mock_ddp_response(hass, MOCK_STATUS_PLAYING, mock_data)
+
+    mock_state = hass.states.get(mock_entity_id).state
+
+    # Ensure that data is not fetched.
+    assert not mock_fetch.mock_calls
+
+    assert mock_state == STATE_PLAYING
+
+    assert len(mock_device.source_list) == 1
+    assert mock_device.source_list[0] == MOCK_TITLE_NAME
+    assert mock_device.media_content_id == MOCK_TITLE_ID
+    assert mock_device.media_title == MOCK_TITLE_NAME
+    assert mock_device.media_image_url == MOCK_TITLE_ART_URL
+    assert mock_device.media_content_type == MOCK_TITLE_TYPE
+
+
+async def test_device_info_is_set_from_status_correctly(hass):
+    """Test that device info is set correctly from status update."""
+    mock_entity_id = await setup_mock_component(hass)
+    mock_device = hass.data[PS4_DATA].devices[0]
+
+    await mock_ddp_response(hass, MOCK_STATUS_OFF)
+    await mock_device.async_get_device_info(MOCK_STATUS_OFF)
```

Comment:
`async_get_device_info` is called via `async_update` which can be triggered by moving time forward.